src/watcher/watcher.worker.ts

```diff
@@ -67,6 +89,46 @@ class Parser {
         this.queue.push(...paths);
         if (!this.parsing) this.parse();
     }
+    processContent(path: string, content: string) {
+        if (this.debug)
+            console.debug(`Process Content: ${path}`)
+        let fileDetails = this.pendingContentProcessing.get(path)
+        let statBlock = this.findFirstStatBlock(content);
+        if (statBlock) {
+            if (this.debug)
+                console.debug(`StatBlocks: ${JSON.stringify(statBlock)}`)
+            const monster: Monster = Object.assign({}, YAML.parse(statBlock), {
+                note: path,
+                mtime: fileDetails.mtime
+            });
+            if (this.debug)
+                console.debug(`MONSTER: ${JSON.stringify(monster)}`)
+            this.processMonster(monster, fileDetails)
+        }
+
+        this.pendingContentProcessing.delete(path)
+        this.checkForWorkComplete();
+    }
+
+    findFirstStatBlock(content: string) : string {
+        let blocks : string[] = []
+        let parts = content.split("```")
+        parts.shift()
+        while (parts.length > 0) {
```

Comment:
I'm not a huge fan of while loops. Can you refactor this to a match and for loop against /^\`\`\`statblock([\s\S]+?)^\`\`\`/gm
?

Reply:
I can simplify this drastically by not having a loop, since it currently only returns one stat block if you'd prefer that for now.  But I had tested the basic algorithm for multiple stat blocks and did minimal changes since your Discord comment made it sound like that might be future direction.

Reply:
I also think the regex against the match you have given is not correct, quite correct, though it can be massaged into correctness.  In particular, the trailing `^` looks wrong to me.  I'd probably used `.` in lieu of [\s\S], and you need to allow white space (but not newlines)  before `statblock` with a newline after it.  If were go this way, perhaps one array of code blocks that is filtered to stat  blocks.

Reply:
The second `^` is used to anchor the second code block back tick set to a newline. You have to use the `m` flag for that. 

`[\s\S]` includes new lines. If you’d rather use the dot, you have to turn on the dot all flag to capture new lines as well.

Reply:
And you've hit on exactly why I dislike multi-line regex's :). I'll make it loop less for now.

Reply:
It is now loopless.  I also adjusted some debug logging statements, including adding the TTRPG: prefix.